### PR TITLE
[Vision] Fix #1179: fix batch size, add unittest

### LINKED ISSUE #1179: TypeError: '<' not supported between instances of 'int' and 'Categorical'
I think after PR #1164 I can use 'medium_quality_faster_inference' preset because previously I am getting error #1141. But now I am getting a new error. 
Version: 0.2.1b20210612
Colab notebook
https://colab.research.google.com/drive/1SshDbvdCSzvRpxDc5qRsvxM03NfKEUqp?usp=sharing

### PR DESCRIPTION
**Pull request by zhreshold**
*Issue #, if available:*
#1179

*Description of changes:*

- Fix batch size, catch invalid types
- Add unittests for presets

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: vision/tests/unittests/test_image_classification.py [hunk 1]
```diff
@@ -64,3 +64,11 @@ def test_invalid_image_dataset():
     df = ImagePredictor.Dataset.from_csv(os.path.join(invalid_test, 'train.csv'), root=os.path.join(invalid_test, 'train_images'))
     predictor = ImagePredictor(label="labels")
     predictor.fit(df, df.copy(), time_limit=60)
+
+
+def test_image_predictor_presets():
+    ImagePredictor = Task
+    train_dataset, _, test_dataset = ImagePredictor.Dataset.from_folders('https://autogluon.s3.amazonaws.com/datasets/shopee-iet.zip')
+    for preset in ['best_quality', 'good_quality_fast_inference', 'medium_quality_faster_train', 'medium_quality_faster_inference']:
```
Comment (Innixma) [inline vision/tests/unittests/test_image_classification.py:72]: It's probably only practical to do medium quality. best_quality would take many hours as currently implemented. time_limit also has to be specified and be a low value (such as 60/120s)

**Comment by szha** [comment]
Job PR-1183-8 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1183/8/index.html

**Comment by szha** [comment]
Job PR-1183-9 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1183/9/index.html

**Comment by szha** [comment]
Job PR-1183-10 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1183/10/index.html

**Comment by zhreshold** [comment]
@Innixma Can you take a look and merge this?

**Comment by szha** [comment]
Job PR-1183-11 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1183/11/index.html

**Review by sxjscience — APPROVED** [review]
LGTM

**Review by Innixma — APPROVED** [review]

**Pull request merged**